### ISSUE TITLE
Circular structure error when serializing consumer object

### ISSUE THREAD

**Issue by Clarity-89** [maykinmedia/consumerjs#36]
`__consumer__` [property](https://github.com/maykinmedia/consumerjs/blob/master/src/abstract-consumer-object.js#L17) on consumer object causes circular structure error when manually serializing it (for example when saving data to local storage).

Error:
```
Uncaught TypeError: Converting circular structure to JSON
    --> starting at object with constructor 'CourseConsumer'
    |     property 'client' -> object with constructor 'e'
    --- property 'consumer' closes the circle
```

**Comment by svenvandescheur** [comment]
Yes, there are internal fields set on objects to keep track of state. There is a (semi private) API to return an object without these items: https://github.com/maykinmedia/consumerjs/blob/master/src/utils.js#L75.

This function simply returns an object without keys listen `unserializableFields` on the `Consumer` instance.  It should be provided by the package as part of "utils" export: https://github.com/maykinmedia/consumerjs/blob/master/src/index.js#L19. And be usable like this:

`JSON.stringify(excludeUnserializableFields(course));`

Does this help you? I'm happy to accept a pull request which exposes this as shortcut as `AbstractConsumerObject.asJSON()`.

**Comment by svenvandescheur** [comment]
Hi @Clarity-89 I've merged your pull request and published it to NPM as version 3.2.0. Thanks!

**Comment by Clarity-89** [comment]
@svenvandescheur awesome! Thanks for such a quick response 🎉
